news/blog: Dashboard Landing Page (#509)
ceph/ceph.io: src/en/news/blog/2023/landing-page/index.md

---

> title: Dashboard Landing Page

Author: You are right, it fits way better, thanks for the suggestion!

date: 2023-02-14
author: Pedro Gonzalez Gomez
image: "image.jpg"
tags:
  - "Ceph"
  - "Dashboard"
---

## Ceph Dashboard and The Landing page

> Ceph Dashboard is a built-in web-based management and monitoring application through which you can inspect and administer various aspects and resources within a Ceph cluster. It is implemented as a Ceph Manager Daemon module. Learn more about it [here](https://docs.ceph.com/en/quincy/mgr/dashboard/#overview).

Reviewer: ```suggestion
The Ceph Dashboard is a web-based management and monitoring application through which you can inspect and control various aspects of a Ceph cluster. It is an integral part of Ceph Manager Daemon module. Learn more about it [here](https://docs.ceph.com/en/quincy/mgr/dashboard/#overview).
```

> The landing page is the home page of the dashboard, this one displays important metrics such as the overall cluster status, performance, and capacity. It provides real-time updates on changes in the cluster and allows quick access to other sections of the dashboard.
> 
> As the Ceph Dashboard continues to add new features and capabilities, the landing page has been updated to keep pace. We are excited to present the updated landing page:

Reviewer: ```suggestion
The landing page of Ceph Dashboard serves as the homepage and features metrics such as the overall cluster status, performance, and capacity. It provides real-time updates on any changes in the cluster, allowing easy navigation to other dashboard sections.

As the Ceph Dashboard adds new features and capabilities, the landing page has been updated to keep pace. We are excited to present the latest version of the landing page:
```

![Landing page](assets/landing-page.png)

## Goals

> The goal of this project is to optimize the user experience of the landing page. This will be achieved through a combination of preserving existing features, introducing new ones, presenting information in a clear and organized manner, and crafting a user-friendly interface. By following these guidelines, we aim to make the landing page a powerful tool for monitoring the cluster with ease.

Reviewer: ```suggestion
The goal of this project is to enhance the user experience of the landing page. This will be achieved through a combination of preserving existing features, introducing new ones, presenting information in a clear and organized manner, and crafting a user-friendly interface. By following these guidelines, we aim to make the landing page a powerful tool for monitoring the cluster with ease.
```

To accomplish this, the main focus of development was on:

- **Maintain Existing Features**
- **Enhance Additional Features**
- **Maintain Structured Content**
- **User-Centered Design**

## Components

In this section, we will take a closer look at the individual components or "cards" that make up the layout of this dashboard.

### Details card

![details-card](assets/details-card.png)

> The Details Card provides an overview of the cluster configuration, displaying various critical aspects of the cluster.

Reviewer: ```suggestion
The Details Card provides an overview of the cluster configuration, highlighting essential cluster attributes.
```

### Status card

![status-card](assets/status-card.png)
The Status Card provides information about the overall health of the cluster and presents it with a visual icon. Additionally, alerts regarding the cluster's state are grouped into danger and warning categories for easy interpretation.

The view can be expanded to allow for viewing and scrolling through alerts, organized by status, danger and warning.

![status-card-open](assets/status-card-open.png)

### Capacity card

![capacity-card-blue](assets/capacity-card-blue.png)

> The Capacity Card displays the total and current capacity of the cluster using a double doughnut chart. The outer ring of the chart represents the nearfull and full-ratios thresholds configured for the OSDs, while the inner blue ring indicates the used capacity. This chart helps to keep track of the total used capacity of the cluster and monitor if it remains below the configured thresholds.

Reviewer 2: ```suggestion
The Capacity Card displays the total and used capacity of the cluster using a double donut chart. The outer ring of the chart represents the near-full and full-ratio thresholds configured for the OSDs, while the inner blue ring shows the used capacity. This chart helps to keep track of the total used capacity of the cluster and monitor whether it remains below the configured thresholds.
```

The used capacity, displayed with the inner blue ring, will turn yellow and red after passing the warning and danger thresholds respectively:

![capacity-card-yellow](assets/capacity-card-yellow.png)

### Inventory card

![inventory-card](assets/inventory-card.png)

> The Inventory Card serves as an inventory of the cluster and allows you to keep track of all assets within the cluster. You can directly access each item from this card by clicking on its link, which will redirect you to the corresponding page.

Reviewer 2: I don't have any issues with the 2nd-person style (vs. "allows users to keep track..."), and in fact I think that it suits the blog style and makes it closer to reader. In that case, I'd review all the sentences where you wrote "allows (sb) to" and change them to "allows you" (same with other causative verbs).

### Cluster utilization card

![cluster-utilization-card](assets/cluster-utilization-card.png)

> The Cluster Utilization Card is populated by various charts, from top to bottom:

Reviewer 2: In other card you used this first sentence to describe the goal of the card, so:
```suggestion
The Cluster Utilization Card shows you the current performance of your cluster:
```

- **Used capacity:** Total capacity used of our cluster. The maximum value of the chart is the maximum capacity of the cluster.

> - **IOPS (Input/Output Operations Per Second):** It represents the number of read and write operations that the OSDs are performing in a single second

Reviewer 2: ```suggestion
- **IOPS (Input/Output Operations Per Second):** It represents the number of read and write operations that the cluster is handling.
```

> - **Latency:** Refers to the amount of time that it takes for the OSDs to be processed and for a response to be received.

Reviewer 2: ```suggestion
- **Latency:** Refers to the amount of time that it takes to process a read or a write request.
```

> - **Client throughput:** It measures the amount of data that can be read or written by the client to the pool in a given period of time.

Reviewer 2: ```suggestion
- **Client throughput:** It measures the amount of data that clients read or write to the cluster.
```

> - **Recovery Throughput:** It measures the rate at which the OSDs can recover data from a backup.

Reviewer 2: ```suggestion
- **Recovery Throughput:** similar to the above, but for cluster recovery reads and writes.
```

> The data is directly retrieved from Prometheus and allows you to set up different timeframes ranging from 5 minutes to 7 days.
> All the charts are updated on real-time, so we can keep track of the different operations of our cluster.

Reviewer 2: BTW [this tool](https://www.deepl.com/write) is really helpful:
```suggestion
This data is retrieved directly from Prometheus and allows you to set up different timeframes, ranging from 5 minutes to 7 days.
All the charts are updated in real-time, so you can keep track of the different operations of your cluster.
```

Moreover, by hovering over any graph, you can quickly access its values - a tooltip will appear displaying the time and value at that moment.

> On the left we can see the name of the respective chart and below it the current values for the different graphs. We can hover over them to distinguish between read and write operations.

Reviewer 2: ```suggestion
On the left side you can find the name of every chart and below that the latest value. You can move the mouse over them to distinguish between read and write operations.
```
